xray/commands/scan.go

```diff
@@ -235,37 +247,31 @@ func getAddTaskToProducerFunc(producer parallel.Runner, errorsQueue *clientutils
 	}
 }
 
-func (scanCmd *ScanCommand) performScanTasks(fileConsumer parallel.Runner, indexedFileConsumer parallel.Runner, resultsArr [][]*services.ScanResponse) (bool, error) {
+func (scanCmd *ScanCommand) performScanTasks(fileConsumer parallel.Runner, indexedFileConsumer parallel.Runner) {
 	go func() {
 		// Blocking until consuming is finished.
 		fileConsumer.Run()
-		// After all files has been indexed, The seconde producer notifies that no more tasks will be produced.
+		// After all files have been indexed, The second producer notifies that no more tasks will be produced.
+		indexedFileConsumer.Done()
+	}()
+	// Blocking until consuming is finished.
+	indexedFileConsumer.Run()
+}
+
+func (scanCmd *ScanCommand) checkScanPassed(fileConsumer parallel.Runner, indexedFileConsumer parallel.Runner) {
```

Comment:
It looks like this method is unused and should be deleted. It is a duplicate of the ```performScanTasks```.